### PR TITLE
review: Use official qodana github action

### PR DESCRIPTION
**Pull request by MartinWitt**
Like ever enterprise project, the developers make horrible decisions at someday and remove the usability.
The Qodana action no longer supports the options we gave it. We need to provide these options as a single string. This is a result of the change in the action using Qodana CLI. Qodana only gave a warning in the log for this behavior, and the last runs since this change could have produced wrong results.
Maybe someone with more advanced YAML skills has a solution for this hideous long string?

### CODE REVIEW

File: .github/workflows/tests.yml
Comment (slarse) [inline .github/workflows/tests.yml:164]: Don't remove the commit pins. Pinning on tag is not safe as they can move. But most prominently it leads to unexpected failures when devs introduce breaking changes and move tags to those.

File: .github/workflows/tests.yml
Comment (slarse) [inline .github/workflows/tests.yml:161]: You could split this line on multiple lines (if GHA supports this YAML syntax, but I think it should):

```suggestion
          args: >-
              --linter,
              ${{ env.QODANA_LINTER }},
              project-dir,
              ${{ github.workspace }}/pull_request,
              changes,
              true,
              fail-threshold,
              0
```

File: src/main/java/spoon/support/compiler/ZipFolder.java
Comment (slarse) [inline src/main/java/spoon/support/compiler/ZipFolder.java:44]: Guessing this is just to trigger an error?

**Review by slarse — CHANGES_REQUESTED** [review]

File: src/main/java/spoon/support/compiler/ZipFolder.java
Comment (MartinWitt) [reply src/main/java/spoon/support/compiler/ZipFolder.java:44]: Yes

**Comment by MartinWitt** [comment]
This ports Qodana to the new style. We fail the CI if any problems are found. If we see this is too strict, we can disable it.

**Comment by I-Al-Istannen** [comment]
I'd be happy to merge if you could quickly test whether it really fails the check. There were significant changes since the last time this was tested in this PR, if I did not miss anything.

**Comment by slarse** [comment]
@MartinWitt :point_up: conflict :o

I've been incredibly busy with other things for the past couple of weeks and feel a bit out of the loop, is there anything keeping us from merging this (except for the conflict)?

**Comment by I-Al-Istannen** [comment]
> I've been incredibly busy with other things for the past couple of weeks and feel a bit out of the loop, is there anything keeping us from merging this (except for the conflict)?

Your guess is as good as mine, but I don't think so. I would really love to get this merged though, as the current pinned hash is now expired and qodana runs fail with:
```
This IDE build has expired. Please download a new build from JetBrains official site
✗  Qodana exited with code 7
```
See https://github.com/INRIA/spoon/actions/runs/3505578524/jobs/5871965639.

**Comment by MartinWitt** [comment]
I fixed the conflict. Shall I push some bad to showcase Qodana is working?

**Comment by I-Al-Istannen** [comment]
> I fixed the conflict. Shall I push some bad to showcase Qodana is working?

I'd appreciate it :)

**Comment by MartinWitt** [comment]
@I-Al-Istannen CI is red as wanted.

**Comment by I-Al-Istannen** [comment]
Looks good to me :)

**Review by I-Al-Istannen — APPROVED** [review]

**Comment by slarse** [comment]
Looks like CI extra checks are failing as snapshots can't be fetched from OW2 repository :/

**Comment by I-Al-Istannen** [comment]
@slarse Fixed now, I think you can review/merge it. Your negative review is greying out the merge button — though I might be able to overrule you.

**Review by slarse — APPROVED** [review]

**Comment by slarse** [comment]
@I-Al-Istannen I'd be more comfortable if you merged as you were the last to look at it, I've approved so there shouldn't be any problems now.

**Comment by I-Al-Istannen** [comment]
> I'd be more comfortable if you merged as you were the last to look at it, I've approved so there shouldn't be any problems now.

A wise decision, as qodana no longer seems to perform delta comparisons. I was a bit suspicious but the test here seems to have worked so I accepted it. I did *not* realize that @MartinWitt was actually **NOT testing his changes here** — the target was not `pull_request_target`. One day I'll remember all the pitfalls of Github Actions.

So, @MartinWitt could you make a new PR with actual delta checks?

**Comment by slarse** [comment]
> A wise decision, as qodana now failed on master :)

Jackpot.

**Pull request merged**
